src/core/core.datasetController.js

```diff
@@ -473,23 +473,22 @@ helpers.extend(DatasetController.prototype, {
 		const me = this;
 		const {_cachedMeta: meta, _data: data} = me;
 		const {iScale, vScale, _stacked} = meta;
-		const parsing = resolve([me.getDataset().parsing, me.chart.options.parsing, true]);
+		const parsing = me._parsing = resolve([me.getDataset().parsing, me.chart.options.parsing, true]);
 		let offset = 0;
 		let i, parsed;
 
-		if (parsing === false) {
-			parsed = data;
-			offset = start;
-		} else if (helpers.isArray(data[start])) {
-			parsed = me._parseArrayData(meta, data, start, count);
-		} else if (helpers.isObject(data[start])) {
-			parsed = me._parseObjectData(meta, data, start, count);
-		} else {
-			parsed = me._parsePrimitiveData(meta, data, start, count);
-		}
+		if (parsing !== false) {
```

Comment:
it looks like `updateStacks` might not work in the case that `parsing === false`

Reply:
`parsing: false` does not affect doughtnut at all (its overwriting the parsing and will probably not have enough data entries to make any difference anyway).
I fixed the optimized internal stuff to work.

Also added a test for the data synchronization that I believe #6814 fails in.